meson.build

```diff
@@ -10,7 +10,7 @@ project('guard', ['cpp', 'c'],
           'warning_level=3',
           'default_library=shared',
           'werror=true',
-          'cpp_std=c++17'
+          'cpp_std=c++20'
```

Comment:
any specific reason to use c++20? (Good to use the latest compiler, just trying to see "are you using any c++20 feature in this patch, nothing hitting to my eye :) ) 

JS: If yes, please add the details in the commit message

Reply:
static guard_log_settings g_guard_log_settings = {
.log_level = GUARD_DEBUG,
.log_func = guard_log_default,
.log_func_priv_data = nullptr,
};

this kind of initialization calls for c++20

Reply:
FYI, what you mentioned is not a C++ feature, it is a C feature "Designated Initializers" that is enabled in the C99. FYR: https://gcc.gnu.org/onlinedocs/gcc/Designated-Inits.html